### PR TITLE
Support MLI 4E

### PR DESCRIPTION
**Pull request by Josephu**
Hi, not sure if this is something you would consider supporting?

I need it for my integration, so I am looking to see if I can add this feature into ISOSIM. Would you like to introduce it to the main code? Or do you prefer me do it all in my fork of ISOSIM and not impacting your master branch? I am more than happy to close this PR if you don't want this to be introduced.

Note that I have already forked your ISOSIM branch to my own for development, so I am more than happy to do my own thing, as well as send my changes if you are keen to support this.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module go
+module github.com/Josephu/go
```
Comment (Josephu) [inline go.mod:1]: At the moment, I had to change this to make it work for my ISOSIM build, I can change this back if you are ok with merging the code back.

File: net/net_cat.go [hunk 2]
```diff
@@ -69,15 +70,23 @@ func (nt *NetCatClient) ReadNextPacket() ([]byte, error) {
...
+	tmp := make([]byte, mliByteLength)
 	_, err := nt.conn.Read(tmp)
 	if err != nil {
 		//if connection has been closed
 		//return
 		return nil, err
 	}
 
-	msgLen := binary.BigEndian.Uint16(tmp)
+	msgLen := uint32(binary.BigEndian.Uint16(tmp))
```
Comment (Josephu) [inline net/net_cat.go:85]: I forced the variable to be unsigned int32, so it can fit both use cases.

**Comment by rkbalgi** [comment]
Hi @Josephu - I'm certainly open to incorporating your changes!. I will take a quick look today. Thanks!

File: go.mod (hunk 1 above)
Comment (rkbalgi) [reply go.mod:1]: No problem, I'll take care of it. I will include 4I and some tests as well.

**Pull request merged**